File: DoctorFei_iOS/DoctorFei_iOS/Controller/Main/AgendaTimeScheduleViewController.m
//
//  AgendaTimaScheduleViewController.m
//  DoctorFei_iOS
//
//  Created by GuJunjia on 15/1/13.
//
//

#import "AgendaTimeScheduleViewController.h"
#import "DoctorAPI.h"
#import <MBProgressHUD.h>
@interface AgendaTimeScheduleViewController()
@property (weak, nonatomic) IBOutlet UIImageView *scheduleTableBackImage;
@property (nonatomic, copy) NSDictionary *scheduleMap;
@property (nonatomic, copy) NSArray *scheduleDay;
@end
@implementation AgendaTimeScheduleViewController
{
    MBProgressHUD *hud;
}

-(void)viewDidLoad
{
    [super viewDidLoad];
    self.scheduleTableBackImage.layer.cornerRadius = 5.0;
    self.scheduleTableBackImage.layer.masksToBounds = YES;
    self.scheduleMap = @{@"Monday_AM": @11,
                         @"Monday_PM": @12,
                         @"Tuesday_AM": @13,
                         @"Tuesday_PM": @14,
                         @"Wednesday_AM":@15,
                         @"Wednesday_PM":@16,
                         @"Thursday_AM":@17,
                         @"Thursday_PM":@18,
                         @"Friday_AM":@19,
                         @"Friday_PM":@20,
                         @"Saturday_AM":@21,
                         @"Saturday_PM":@22,
                         @"Sunday_AM":@23,
                         @"Sunday_PM":@24};
    
}
-(void)viewWillAppear:(BOOL)animated
{
    [super viewWillAppear:animated];
//    [self updateSchedule];
    [self loadSchedule];
}
-(void)loadSchedule
{
    NSNumber *doctorId = [[NSUserDefaults standardUserDefaults]objectForKey:@"UserId"];
    NSDictionary *params = @{
                             @"doctorid": doctorId
                             };
    [DoctorAPI getDoctorScheduleWithParameters:params success:^(AFHTTPRequestOperation *operation, id responseObject) {
        NSLog(@"%@",responseObject);
        for (id str in [[responseObject firstObject] allKeys]){
            NSNumber *tag = nil;
            if ((tag =[self.scheduleMap objectForKey:str])){
                UIButton *btn = (UIButton *)[self.view viewWithTag:[tag integerValue]];
                btn.selected = [responseObject[0][str] integerValue] == 1;
            }
        }
    } failure:^(AFHTTPRequestOperation *operation, NSError *error) {
        NSLog(@"%@", error.localizedDescription);
    }];
}

-(void)updateSchedule
{
    NSNumber *doctorId = [[NSUserDefaults standardUserDefaults]objectForKey:@"UserId"];
    NSDictionary *params = @{
                             @"doctorid": doctorId,
                             @"Monday_AM": [self scheduleDataWithKey:@"Monday_AM"],
                             @"Monday_PM": [self scheduleDataWithKey:@"Monday_PM"],
                             @"Tuesday_AM": [self scheduleDataWithKey:@"Tuesday_AM"],
                             @"Tuesday_PM": [self scheduleDataWithKey:@"Tuesday_PM"],
                             @"Wednesday_AM": [self scheduleDataWithKey:@"Wednesday_AM"],
                             @"Wednesday_PM": [self scheduleDataWithKey:@"Wednesday_PM"],
                             @"Thursday_AM": [self scheduleDataWithKey:@"Thursday_AM"],
                             @"Thursday_PM": [self scheduleDataWithKey:@"Thursday_PM"],
                             @"Friday_AM": [self scheduleDataWithKey:@"Friday_AM"],
                             @"Friday_PM": [self scheduleDataWithKey:@"Friday_PM"],
                             @"Saturday_AM": [self scheduleDataWithKey:@"Saturday_AM"],
                             @"Saturday_PM": [self scheduleDataWithKey:@"Saturday_PM"],
                             @"Sunday_AM": [self scheduleDataWithKey:@"Sunday_AM"],
                             @"Sunday_PM": [self scheduleDataWithKey:@"Sunday_PM"]
                             };
    [DoctorAPI updateDoctorScheduleWithParameters:params success:^(AFHTTPRequestOperation *operation, id responseObject) {
        NSLog(@"%@",responseObject);
        NSDictionary *dic = [responseObject firstObject];
        hud = [MBProgressHUD showHUDAddedTo:self.view.window animated:YES];
        hud.mode = MBProgressHUDModeCustomView;
        UIImageView *imageView = [[UIImageView alloc] initWithImage:[UIImage imageNamed:@"success_pic.png"]];
        if([dic[@"state"] integerValue] == 1)
        {
            hud.customView = imageView;
        }
        else
        {
        }
        NSLog(@"%@",dic[@"msg"]);
        hud.dimBackground = YES;
        hud.labelText = dic[@"msg"];
        [hud hide:YES afterDelay:2.0];
    } failure:^(AFHTTPRequestOperation *operation, NSError *error) {
        NSLog(@"%@", error.localizedDescription);
    }];
}
-(IBAction)scheduleSelected:(UIButton *)sender{
    sender.selected = !sender.selected;
}
- (IBAction)confirm:(UIBarButtonItem *)sender {
    [self updateSchedule];
}
- (NSNumber *)scheduleDataWithKey:(NSString *)str{
    NSNumber *tag = self.scheduleMap[str];
    if (!tag)
        return @0;
    UIButton *btn = (UIButton *)[self.view viewWithTag:[tag integerValue]];
    if (btn.selected)
        return @1;
    else
        return @0;
}
@end

File: DoctorFei_Patient_iOS/DoctorFei_Patient_iOS/Controller/Contact/AgendaTimeScheduleViewController.swift
//
//  AgendaTimeScheduleViewController.swift
//  DoctorFei_Patient_iOS
//
//  Created by shadowPriest on 3/15/15.
//
//

import UIKit

class AgendaTimeScheduleViewController: UIViewController {
    var doctorId = NSNumber()
    
    
    @IBOutlet weak var scheduleTableBackImage: UIImageView!
    var scheduleMap = NSDictionary()

    override func viewDidLoad() {
        super.viewDidLoad()
        self.scheduleTableBackImage.layer.cornerRadius = 5.0;
        self.scheduleTableBackImage.layer.masksToBounds = true;
        self.scheduleMap = [ "Monday_AM": NSNumber(integer: 11),
                         "Monday_PM": NSNumber(integer: 12),
                         "Tuesday_AM": NSNumber(integer: 13),
                         "Tuesday_PM": NSNumber(integer: 14),
                         "Wednesday_AM":NSNumber(integer: 15),
                         "Wednesday_PM":NSNumber(integer: 16),
                         "Thursday_AM":NSNumber(integer: 17),
                         "Thursday_PM":NSNumber(integer: 18),
                         "Friday_AM":NSNumber(integer: 19),
                         "Friday_PM":NSNumber(integer: 20),
                         "Saturday_AM":NSNumber(integer: 21),
                         "Saturday_PM":NSNumber(integer: 22),
                         "Sunday_AM":NSNumber(integer: 23),
                         "Sunday_PM":NSNumber(integer: 24)] as NSDictionary
        
        // Do any additional setup after loading the view.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    override func viewWillAppear(animated: Bool) {
        super.viewWillAppear(animated)
        self.loadSchedule()
    }
    func test(){
        test()
    }
    func testCallByName(bl: Bool,testfunc: @autoclosure ()->Void ){
        if bl {
            println("yes")
        }else {
            println("no")
        }
    }
    func testCallByValue(bl: Bool,testfunc:Void ){
        if bl{
            println("yes")
        }
        else{
            println("no")
        }
    }
    
    @IBAction func backButtonClicked(sender: AnyObject) {
//        testCallByName(true, testfunc: test())
//        testCallByValue(true, testfunc: test())
        self.navigationController?.popViewControllerAnimated(true)
    }
    func loadSchedule() {
        let params = [ "doctorid" : self.doctorId] as NSDictionary
        MemberAPI.getDoctorScheduleWithParameters(params, success: {
            operation, responseObject in
            NSLog("%@", responseObject as NSArray)
            if (responseObject as NSArray).count > 0 {
                for (key,value) in (responseObject? as NSArray).firstObject as NSDictionary{
                    NSLog("%@:%@", key as NSObject,value as NSObject)
                    if let tTag: AnyObject = self.scheduleMap.objectForKey(key){
                        let tag = tTag as NSNumber
                        var btn = self.view.viewWithTag(tag.integerValue)? as UIButton?
                        btn?.selected = (value as NSNumber).integerValue == 1
                    }
                }
            }
            }, failure: {
                operation, error in
                NSLog("%@", error)
        })
    }
    /*
    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
        // Get the new view controller using segue.destinationViewController.
        // Pass the selected object to the new view controller.
    }
    */

}
